sklearn/cross_validation.py

```diff
@@ -1010,14 +1010,19 @@ def _iter_indices(self):
             # Because of rounding issues (as n_train and n_test are not
             # dividers of the number of elements per class), we may end
             # up here with less samples in train and test than asked for.
-            if len(train) < self.n_train or len(test) < self.n_test:
+            if len(train) + len(test) < self.n_train + self.n_test:
                 # We complete by affecting randomly the missing indexes
                 missing_idx = np.where(bincount(train + test,
                                                 minlength=len(self.y)) == 0,
                                        )[0]
                 missing_idx = rng.permutation(missing_idx)
-                train.extend(missing_idx[:(self.n_train - len(train))])
```

Comment:
The crux of the problem is that `self.n_train - len(train)` can be negative if you are unlucky with the rounding, which causes a problem in the slicing below.